actions/storage.js

```diff
@@ -80,13 +80,31 @@ export function actionOnItemsWithPrefix(prefix, action) {
     };
 }
 
-export function storageRehydrate(incoming) {
-    return async (dispatch, persistor) => {
+export function storageRehydrate(incoming, persistor) {
+    return async (dispatch, getState) => {
+        const state = getState();
         persistor.pause();
         Object.keys(incoming).forEach((key) => {
             const storage = {};
             try {
-                storage[key] = JSON.parse(incoming[key]);
+                const value = JSON.parse(incoming[key]);
+                if (value === null) {
+                    storage[key] = {value, timestamp: new Date()};
+                } else if (typeof state.storage.storage[key] === 'undefined') {
+                    if (typeof value.timestamp === 'undefined') {
+                        storage[key] = {value, timestamp: new Date()};
+                    } else {
+                        storage[key] = {value: value.value, timestamp: new Date(value.timestamp)};
+                    }
+                } else if (typeof value.timestamp === 'undefined') {
+                    storage[key] = {value, timestamp: new Date()};
+                } else if (typeof state.storage.storage[key].timestamp === 'undefined') {
+                    storage[key] = {value: value.value, timestamp: new Date(value.timestamp)};
+                } else if (new Date(value.timestamp) > state.storage.storage[key].timestamp) {
```

Comment:
Because the value.timestamp comes from the localForage as a JSON serialized string, so JSON.parse doesn't know about "Date" objects, so, it returns a string representing the Date object. This `new Date(value.timestamp)` return the timestamp to the Date format (which is what is used in the storage).